pennylane/templates/embeddings.py

```diff
@@ -136,15 +141,30 @@ def AmplitudeEmbedding(features, wires):
     Args:
         features (array): Input array of shape ``(2**n,)``
         wires (Sequence[int]): sequence of qubit indices that the template acts on
+        pad (Boolean): controls the activation of the padding option, defaults to ``False``
+        normalize (Boolean): controls the activation of automatic normalization, defaults to ``False``
     """
 
     if not isinstance(wires, Iterable):
         raise ValueError("Wires needs to be a list of wires that the embedding uses; got {}.".format(wires))
 
-    if 2**len(wires) != len(features):
-        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**len(wires), which is {}; "
+    if pad and 2**len(wires) >= len(features):
+        features = np.pad(features, (0, 2**len(wires)-len(features)), 'constant')
+
+    if pad and 2**len(wires) < len(features):
+        raise ValueError("AmplitudeEmbedding with padding requires the size of feature vector to be smaller than 2**len(wires), which is {}; "
```

Comment:
AmplitudeEmbedding _always_ requires a feature vector that is smaller or equal to `2**len(wires)` in length. You can remove the `pad` from the if and move the if up so that it's the second check. 

```suggestion
        raise ValueError("AmplitudeEmbedding requires the size of feature vector to be smaller than 2**len(wires), which is {}; "
```